app/src/main/java/com/github/se/polyfit/viewmodel/qrCode/barcodeViewModel.kt

```diff
@@ -19,17 +24,24 @@ const val REQUIRED_SCAN_COUNT = 3
 const val MIN_BARCODE_LENGTH = 6
 const val MAX_BARCODE_LENGTH = 13
 
+const val TIME_COLOR_BORDER_MILLIS = 2000L
+
 @HiltViewModel
 class BarCodeCodeViewModel @Inject constructor() : ViewModel() {
   private val _listId = MutableLiveData<List<String>>(emptyList())
+  val _isScanned = MutableLiveData(false)
+  private var lastScanTime = System.currentTimeMillis()
```

Comment:
I think you should initialize it to `0L` as it will only be used in addId and will be modified at that point